src/native.c

```diff
@@ -476,7 +476,11 @@ static enum mcrx_error_code mcrx_find_interface(
       family = AF_INET;
       struct sockaddr_in* sp4 = (struct sockaddr_in*)sp;
       sp4->sin_family = AF_INET;
-      sp4->sin_addr = sub->input.addrs.v4.source;
+      if (sub->mnat_entry == NULL) {
```

Comment:
I think this check needs to be looking at the ctx->mnatmap or similar.  If mnat is enabled in the ctx, then all subscriptions in the ctx without a mapping to local addresses are inactive, not just transparently using the global addresses.